extensions/data-transfer/portability-data-transfer-imgur/src/main/java/org/datatransferproject/datatransfer/imgur/photos/ImgurPhotosImporter.java

```diff
@@ -0,0 +1,194 @@
+    }
+    return new ImportResult(ImportResult.ResultType.OK);
+  }
+
+  private void importAlbum(
+      PhotoAlbum album, UUID jobId, TokensAndUrlAuthData authData, TempPhotosData photoData)
+      throws IOException {
+    String description = album.getDescription();
+
+    Request.Builder requestBuilder = new Request.Builder().url(CREATE_ALBUM_URL);
+    requestBuilder.header("Authorization", "Bearer " + authData.getAccessToken());
+
+    FormBody.Builder builder = new FormBody.Builder().add("title", album.getName());
+    if (!Strings.isNullOrEmpty(description)) {
+      builder.add("description", description);
+    }
+    RequestBody formBody = builder.build();
+    requestBuilder.post(formBody);
+
+    Response response = client.newCall(requestBuilder.build()).execute();
+    int code = response.code();
+    if (code >= 200 && code <= 299) {
+      ResponseBody body = response.body();
+      if (body == null) {
+        throw new IOException("Didn't get response body");
+      }
+      Map<String, Object> responseData = objectMapper.readValue(body.bytes(), Map.class);
+
+      String newAlbumId = (String) ((Map<String, Object>) responseData.get("data")).get("id");
+      if (Strings.isNullOrEmpty(newAlbumId)) {
+        throw new IOException("Didn't receive new album id");
+      }
+      // Save new album id so that photos could be added later to this album
+      photoData.addAlbumId(album.getId(), newAlbumId);
+      jobStore.update(jobId, createCacheKey(), photoData);
+    } else {
+      throw new IOException(
+          String.format(
+              "Error occurred in request for %s, code: %s, message: %s",
+              CREATE_ALBUM_URL, code, response.message()));
+    }
+  }
+
+  private void importPhoto(PhotoModel photoModel, UUID jobId, TokensAndUrlAuthData authData)
+      throws IOException {
+    InputStream inputStream = null;
+    String albumId = photoModel.getAlbumId();
+    String imageDescription = photoModel.getDescription();
+
+    if (photoModel.isInTempStore()) {
+      inputStream = jobStore.getStream(jobId, photoModel.getFetchableUrl());
+    } else if (photoModel.getFetchableUrl() != null) {
+      inputStream = new URL(photoModel.getFetchableUrl()).openStream();
+    } else {
+      monitor.severe(() -> "Can't get inputStream for a photo");
+      return;
+    }
+
+    byte[] imageBytes = ByteStreams.toByteArray(inputStream);
+    String imageData = Base64.getEncoder().encodeToString(imageBytes);
+
+    Request.Builder requestBuilder = new Request.Builder().url(UPLOAD_PHOTO_URL);
+    requestBuilder.header("Authorization", "Bearer " + authData.getAccessToken());
+
+    FormBody.Builder builder = new FormBody.Builder().add("image", imageData);
+
+    if (!Strings.isNullOrEmpty(albumId)) {
+      TempPhotosData tempData = jobStore.findData(jobId, createCacheKey(), TempPhotosData.class);
+      // Get previously saved id of imported album
+      String newAlbumId = tempData.lookupNewAlbumId(albumId);
```

Comment:
Is it possible that this album Id doesnt exist?

Reply:
Yes, it can be null for non-album photos, then it will be imported as non-album photo too

Reply:
ah okay so setting album to the null value below in builder.add("album", newAlbumId) won't cause issues? or should we add an ifstatement around it?

Reply:
Sorry, I thought you mentioned the old album id.
It's not likely that id of newly created album will be null if old album exists, as during import it [checks](https://github.com/google/data-transfer-project/blob/563e1d62e3325e75fee96add98eb0ea6bc3e363b/extensions/data-transfer/portability-data-transfer-imgur/src/main/java/org/datatransferproject/datatransfer/imgur/photos/ImgurPhotosImporter.java#L128) that new album is created. But I added [ifstatement](https://github.com/google/data-transfer-project/blob/563e1d62e3325e75fee96add98eb0ea6bc3e363b/extensions/data-transfer/portability-data-transfer-imgur/src/main/java/org/datatransferproject/datatransfer/imgur/photos/ImgurPhotosImporter.java#L164) in case of `newAlbumId` is null for any reason

Reply:
Thanks!